diff --git a/src/mosquitto.c b/src/mosquitto.c
index 09632f2..59717f4 100644
--- a/src/mosquitto.c
+++ b/src/mosquitto.c
@@ -402,15 +402,15 @@ static void listeners__stop(void)
 
 static void signal__setup(void)
 {
-	signal(SIGINT, handle_sigint);
-	signal(SIGTERM, handle_sigint);
+	/* signal(SIGINT, handle_sigint); */
+	/* signal(SIGTERM, handle_sigint); */
 #ifdef SIGHUP
-	signal(SIGHUP, handle_sighup);
+	/* signal(SIGHUP, handle_sighup); */
 #endif
 #ifndef WIN32
-	signal(SIGUSR1, handle_sigusr1);
-	signal(SIGUSR2, handle_sigusr2);
-	signal(SIGPIPE, SIG_IGN);
+	/* signal(SIGUSR1, handle_sigusr1); */
+	/* signal(SIGUSR2, handle_sigusr2); */
+	/* signal(SIGPIPE, SIG_IGN); */
 #endif
 #ifdef WIN32
 	CreateThread(NULL, 0, SigThreadProc, NULL, 0, NULL);
